src/main/java/org/jenkinsci/plugins/workflow/cps/SnippetizerLink.java

```diff
@@ -0,0 +1,200 @@
+    @Nonnull
+    public abstract String getUrl();
+
+    /**
+     * Get the actual URL to use in sidepanel.jelly. If {@link #getUrl()} is not absolute, this will try to get the
+     * current Job context and return a url starting with that job's {@link Job#getUrl()} appended with {@link #getUrl()}.
+     */
+    @Nonnull
+    public final String getDisplayUrl() {
+        String u = getUrl();
+
+        try {
+            if (new URI(u).isAbsolute()) {
+                return u;
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.log(Level.WARNING, "Failed to parse URL for {0}: {1}", new Object[]{u, e});
+            return "";
+        }
+
+        StaplerRequest req = Stapler.getCurrentRequest();
+        if (req == null) {
+            return u;
+        }
+
+        Item i = req.findAncestorObject(Item.class);
+
+        if (i == null) {
+            return u;
+        }
+
+        return req.getContextPath() + "/" + i.getUrl() + u;
+    }
+
+    /**
+     * Get the icon information for the link.
+     */
+    @Nonnull
+    public String getIcon() {
+        return "icon-help icon-md";
+    }
+
+    /**
+     * Get the display name for the link.
+     */
+    @Nonnull
+    public abstract String getDisplayName();
+
+    /**
+     * Check whether the link should target a new window - this defaults to false;
+     */
+    public boolean inNewWindow() {
+        return false;
+    }
+
+    @Extension(ordinal = 1000L)
+    public static class GeneratorLink extends SnippetizerLink {
+        @Override
+        @Nonnull
+        public String getUrl() {
+            return ACTION_URL;
+        }
+
+        @Override
+        @Nonnull
+        public String getIcon() {
+            return "icon-gear2 icon-md";
+        }
+
+        @Override
+        @Nonnull
+        public String getDisplayName() {
+            return Messages.SnippetizerLink_GeneratorLink_displayName();
+        }
+    }
+
+    @Extension(ordinal = 900L)
+    public static class StepReferenceLink extends SnippetizerLink {
+        @Override
+        @Nonnull
+        public String getUrl() {
+            return ACTION_URL + "/html";
+        }
+
+        @Override
+        @Nonnull
+        public String getDisplayName() {
+            return Messages.SnippetizerLink_StepReferenceLink_displayName();
+        }
+    }
+
+    @Extension(ordinal = 800L)
+    public static class GlobalsReferenceLink extends SnippetizerLink {
+        @Override
+        @Nonnull
+        public String getUrl() {
+            return ACTION_URL + "/globals";
+        }
+
+        @Override
+        @Nonnull
+        public String getDisplayName() {
+            return Messages.SnippetizerLink_GlobalsReferenceLink_displayName();
+        }
+    }
+
+    @Extension(ordinal = 700L)
+    public static class OnlineDocsLink extends SnippetizerLink {
+        @Override
+        @Nonnull
+        public String getUrl() {
+            return "https://jenkins.io/doc/pipeline/";
```

Comment:
BTW we should be using a redirect for this.

Reply:
+1

Reply:
@jglick Not sure what you mean?

Reply:
and fwiw, all I'm doing here is replicating the existing behavior of the rendered HTML, so if that should be changed, that should probably be handled separately from this.

Reply:
Yes it could be handled separately, merely noting it as an aside. By redirect I mean `https://jenkins.io/redirect/some-short-phrase-here/`, the target of which can be controlled by a simple PR to `jenkins-infra/jenkins.io`, so that you can rearrange web properties later without touching the product.

Reply:
Aaaah, good idea.